prog/dftb+/api/mm/mmapi.F90

```diff
@@ -84,6 +84,48 @@ module dftbp_mmapi
 
 contains
 
+
+  !> Return the version string for the current DFTB+ build
+  subroutine getDftbPlusBuild(version)
+
+    !> Version string for DFTB+
+    character(:), allocatable, intent(out) :: version
+
+    version = '${RELEASE}$'
+
+  end subroutine getDftbPlusBuild
+
+
+  !> Returns the DFTB+ API version
+  subroutine getDftbPlusApi(major, minor, patch)
+
+    !> Major version number
+    integer, intent(out) :: major
+
+    !> Minor version number
+    integer, intent(out) :: minor
+
+    !> patch level for API
+    integer, intent(out) :: patch
+
+    character(*), parameter :: version = '${APIVERSION}$'
+    character(len=256) :: majorS, minorS, patchS
+    integer :: iPos, jPos
+
+    ! parse a string of the form 'x.y.z'
+    iPos = index(version, '.')
+    jPos = index(version, '.', back=.true.)
+
+    majorS = version(1:iPos-1)
+    minorS = version(iPos+1:jPos-1)
+    patchS = version(jPos+1:)
+    read(majorS, *) major
+    read(minorS, *) minor
+    read(patchS, *) patch
```

Comment:
This part is quite complicated to parse the API version string and assumes a lot about the `APIVERSION` preprocessor, which offers various possibilities for breakage (`x.y`, `x.y.z-a1`). Failure in this part of the API happens to be always fatal, since it cannot be recovered due to missing guards for the `read`.

This version here is of course fine, but it can be easily broken if one is not cautious about the exact implementation details of the API function which ties in directly with CMake build system.

Maybe I'm a bit overcautious here.

Reply:
The api version number is a flat checked in file at the moment. We should discuss at the next meeting with @aradi.